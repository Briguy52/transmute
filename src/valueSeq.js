import { Iterable, Seq } from "immutable";

/**
 * Get a Seq of the values in `value`.
 */
export default function valueSeq<V>(
  value: Array<V> | Iterable<any, V> | { [key: any]: V }
) {
  if (!Iterable.isIterable(value)) {
    value = Seq(value);
  }
  return value.valueSeq();
}
