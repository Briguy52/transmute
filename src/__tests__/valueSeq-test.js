import { Map } from 'immutable';
import valueSeq from '../valueSeq';

describe('transmute/valueSeq', () => {
  it('gets the values from a Map', () => {
    expect(valueSeq(Map({ one: 1, two: 2, three: 3 }))).toMatchSnapshot();
  });

  it('gets the values from an Array', () => {
    expect(valueSeq([1, 2, 3])).toMatchSnapshot();
  });

  it('gets the values from an Object', () => {
    expect(valueSeq({ one: 1, two: 2, three: 3 })).toMatchSnapshot();
  });
});
