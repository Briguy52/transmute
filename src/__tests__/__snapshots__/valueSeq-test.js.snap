// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transmute/valueSeq gets the values from a Map 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`transmute/valueSeq gets the values from an Array 1`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`transmute/valueSeq gets the values from an Object 1`] = `
Array [
  1,
  2,
  3,
]
`;
